### ISSUE TITLE
Project snapshots

### ISSUE THREAD

**Issue by kazagkazag** [LLK/scratch-gui#777]
### Expected Behavior

Scratch should expose action to create snapshots of projects. Snapshots would be available in some kind of timeline/history panel, where author could select snapshot and "checkout" specific version of the project from the past. It could work like git history, but be more linear (one timeline, no branches) or Photoshop history panel.

### Actual Behavior

There is no such option...

### Why ?

1. It would be much easier for authors to create application with changes history. We can go back and fix some code, delete next snapshots and start from that point. It will be easier than revert all changes manually.
2. Snapshots would be great for Code Clubs community. There is a lot of instructions for Scratch with screenshots of the created application. But when we want to translate instruction into some other language than English, we have to recreate entire application step by step, because sometimes set of variables or scripts are changed in next steps. With snapshots we could simply checkout specific snapshot and take screenshot :)

**Comment by kyleplo** [comment]
Yeah, this sounds like a good idea. You could use _`canvasObj`_`.toDataUrl()`

**Comment by towerofnix** [comment]
@kyleplo That's for taking screenshot images (you know, PNGs) of a canvas - this issue discusses something more like a timeline-based history of code, sprites, etc where you could go back and forth to view old versions of the project.

**Comment by kazagkazag** [comment]
I could start contributing from that issue, but I don't see any contribution guide here? What should I do to develop such feature? From whom should I talk about possible solutions?

**Comment by kazagkazag** [comment]
Any clue ? :)

**Comment by thisandagain** [comment]
@kazagkazag This requires discussion before we will consider implementation or mark as "help wanted". Thanks.

**Comment by mraerino** [comment]
The WebGL Renderer has an option to get a PNG of the stage:
https://github.com/LLK/scratch-render/blob/c9f86ef53bd6633fbb21ae7917b6223d809bebf2/src/RenderWebGL.js#L1798
